src/explore-education-statistics-common/src/components/ModalConfirm.tsx

```diff
@@ -1,41 +1,52 @@
 import ButtonGroup from '@common/components/ButtonGroup';
+import Button from '@common/components/Button';
+import Modal from '@common/components/Modal';
 import useMountedRef from '@common/hooks/useMountedRef';
 import useToggle from '@common/hooks/useToggle';
-import React, { ReactNode } from 'react';
-import Button from './Button';
-import Modal from './Modal';
+import React, { ReactNode, useEffect } from 'react';
 
 interface Props {
   children?: ReactNode;
   className?: string;
-  closeText?: string;
+  cancelText?: string;
   confirmText?: string;
   open?: boolean;
-  onConfirm(): void;
-  onCancel?(): void;
-  onExit(): void;
   showCancel?: boolean;
   title: string;
+  triggerButton?: ReactNode;
   underlayClass?: string;
+  onCancel?(): void;
+  onConfirm(): void;
+  onExit?(): void;
 }
 
 const ModalConfirm = ({
+  cancelText = 'Cancel',
   children,
   className,
   confirmText = 'Confirm',
-  closeText = 'Cancel',
-  open,
-  onConfirm,
-  onExit,
-  onCancel = onExit,
+  open: initialOpen = false,
   showCancel = true,
   title,
+  triggerButton,
   underlayClass,
+  onCancel,
+  onConfirm,
+  onExit,
 }: Props) => {
   const isMounted = useMountedRef();
   const [isDisabled, toggleDisabled] = useToggle(false);
+  const [open, toggleOpen] = useToggle(initialOpen);
 
-  const handleAction = (callback: () => void) => async () => {
+  useEffect(() => {
+    toggleOpen(initialOpen);
+  }, [initialOpen, isMounted, toggleOpen]);
```

Comment:
Shouldn't need `isMounted` here. It's also a ref, so changing it doesn't trigger a re-render anyway.